screenpipe-vision/src/ui_monitoring_macos.swift

```diff
@@ -29,6 +29,20 @@ struct WindowState {
     }
 }
 
+struct UIFrame: Encodable {
+    let window: String
+    let app: String
+    let text_output: String
+    let initial_traversal_at: String
+}
+
+func uiFrameToJson(uiFrame: UIFrame) -> String {
```

Comment:
i wish we stop using JSON

inefficient af 

no?

i think we should use more compact structure and turn into json when necessary

Reply:
I agree that it can be improved.

Reply:
The new implementation reads the `UIFrame` from bytes. @louis030195